ext/mysqli/php_mysqli_structs.h

```diff
@@ -247,26 +247,26 @@ extern void php_mysqli_fetch_into_hash_aux(zval *return_value, MYSQL_RES * resul
 	MYSQLI_RESOURCE *my_res; \
 	mysqli_object *intern = Z_MYSQLI_P(__id); \
 	if (!(my_res = (MYSQLI_RESOURCE *)intern->ptr)) {\
-  		php_error_docref(NULL, E_WARNING, "Couldn't fetch %s", ZSTR_VAL(intern->zo.ce->name));\
-		RETURN_FALSE;\
+		zend_throw_error(NULL, "Couldn't fetch %s", ZSTR_VAL(intern->zo.ce->name));\
+		RETURN_THROWS();\
   	}\
 	__ptr = (__type)my_res->ptr; \
 	if (__check && my_res->status < __check) { \
-		php_error_docref(NULL, E_WARNING, "invalid object or resource %s\n", ZSTR_VAL(intern->zo.ce->name)); \
-		RETURN_FALSE;\
+		zend_throw_error(NULL, "invalid object or resource %s", ZSTR_VAL(intern->zo.ce->name)); \
```

Comment:
I think these error messages should be improved, like: `Invalid object or resource: %s`